### PR TITLE
Make control plane outbound lb configurable

### PR DESCRIPTION
**Pull request by shysank**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

- Adds a field `ControlPlaneOutboundLB` in `AzureCluster` to configure control plane outbound load balancer. This is useful for private clusters when we don't want a public control plane lb created by default
- Works very similar to `NodeOutboundLB`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
Addresses https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1159

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
Make control plane outbound lb configurable
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
@fiunchinho what's the impact of this PR on #894 if any?

**Comment by fiunchinho** [comment]
> @fiunchinho what's the impact of this PR on #894 if any?

I don't think it has any impact, as we only reconcile nat gateways for worker nodes (we don't for the control plane).

**Comment by CecileRobertMichon** [comment]
@shysank PR needs rebase

is this enough to close #1159 ?

**Comment by shysank** [comment]
> is this enough to close #1159 ?

I think so, not sure if we should consider #1394 to be part of it.

**Comment by CecileRobertMichon** [comment]
> I think so, not sure if we should consider #1394 to be part of it.

Let's keep #1394 separate since it has its own issue

File: api/v1alpha4/azurecluster_validation.go [hunk 1]
```diff
@@ -420,6 +422,34 @@ func validateNodeOutboundLB(lb *LoadBalancerSpec, old *LoadBalancerSpec, apiserv
...
+func validateControlPlaneOutboundLB(lb *LoadBalancerSpec, apiserverLB LoadBalancerSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	switch apiserverLB.Type {
+	case Public:
+		if lb != nil {
+			allErrs = append(allErrs, field.Forbidden(fldPath, "Control plane outbound load balancer cannot be set for public clusters."))
+		}
+	case Internal:
+		// Control plane outbound lb can be nil when it's disabled private clusters.
```
Comment (CecileRobertMichon) [inline api/v1alpha4/azurecluster_validation.go:434]:
```suggestion
		// Control plane outbound lb can be nil when it's disabled for private clusters.
```

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Pull request merged**